src/scripts/main.js

```diff
@@ -1,3 +1,449 @@
+
+  paramArr[0] = mergedArr[0] || 0;
+  paramArr[1] = mergedArr[1] || 0;
+  paramArr[2] = mergedArr[2] || 0;
+  paramArr[3] = mergedArr[3] || 0;
+}
+
+function right(paramArr) {
+  const arr = paramArr.filter(item => item !== 0);
+
+  for (let i = arr.length - 1; i >= 0; i--) {
+    const current = arr[i];
+    const prev = arr[i - 1];
+
+    if (current === prev) {
+      arr[i] = current * 2;
+      scoreValue += arr[i];
+      updateScore();
+      arr[i - 1] = 0;
+    }
+  }
+
+  const mergedArr = arr.filter(item => item !== 0);
+
+  while (mergedArr.length < 4) {
+    mergedArr.unshift(0);
+  }
+
+  paramArr[0] = mergedArr[0] || 0;
+  paramArr[1] = mergedArr[1] || 0;
+  paramArr[2] = mergedArr[2] || 0;
+  paramArr[3] = mergedArr[3] || 0;
+}
+
+function moveLeft() {
+  const json = JSON.stringify(table);
+  const copyTable = JSON.parse(json);
+
+  for (const row of table) {
+    left(row);
+  }
+
+  for (let i = 0; i < 16; i++) {
+    const rowIndex = Math.floor(i / 4);
+    const colIndex = i - (rowIndex * 4);
+
+    const tableValue = table[rowIndex][colIndex];
+    const copyTableValue = copyTable[rowIndex][colIndex];
+
+    if (tableValue !== copyTableValue) {
+      appearingOneRandomCell();
+      updateGameFields();
+
+      break;
+    }
+  }
+}
+
+function moveRight() {
+  const json = JSON.stringify(table);
+  const copyTable = JSON.parse(json);
+
+  for (const row of table) {
+    right(row);
+  }
+
+  for (let i = 0; i < 16; i++) {
+    const rowIndex = Math.floor(i / 4);
+    const colIndex = i - (rowIndex * 4);
+
+    const tableValue = table[rowIndex][colIndex];
+    const copyTableValue = copyTable[rowIndex][colIndex];
+
+    if (tableValue !== copyTableValue) {
+      appearingOneRandomCell();
+      updateGameFields();
+
+      break;
+    }
+  }
+}
+
+function moveUp() {
+  const json = JSON.stringify(table);
+  const copyTable = JSON.parse(json);
+
+  for (let j = 0; j < 4; j++) {
+    up([table[0][j], table[1][j], table[2][j], table[3][j]], j);
+  }
+
+  for (let i = 0; i < 16; i++) {
+    const rowIndex = Math.floor(i / 4);
+    const colIndex = i - (rowIndex * 4);
+
+    const tableValue = table[rowIndex][colIndex];
+    const copyTableValue = copyTable[rowIndex][colIndex];
+
+    if (tableValue !== copyTableValue) {
+      appearingOneRandomCell();
+      updateGameFields();
+
+      break;
+    }
+  }
+}
+
+function moveDown() {
+  const json = JSON.stringify(table);
+  const copyTable = JSON.parse(json);
+
+  for (let j = 0; j < 4; j++) {
+    down([table[0][j], table[1][j], table[2][j], table[3][j]], j);
+  }
+
+  for (let i = 0; i < 16; i++) {
+    const rowIndex = Math.floor(i / 4);
+    const colIndex = i - (rowIndex * 4);
+
+    const tableValue = table[rowIndex][colIndex];
+    const copyTableValue = copyTable[rowIndex][colIndex];
+
+    if (tableValue !== copyTableValue) {
+      appearingOneRandomCell();
+      updateGameFields();
+
+      break;
+    }
+  }
+}
```

Comment:
almost same code. difference just in up or down function. combine it in one function which will be accept function up or down as argument